### PR TITLE
[DVM-8260] Handle Server Down Time

### PR DESCRIPTION
**Pull request by winsonsutanto10**
## Description

> Please add your task in ClickUp here.

Task:  [DVM-8260](https://app.clickup.com/t/3857515/DVM-8260)
## What's new?

> Please add if you added new improvement

add CheckFeatureFlagStatus in middleware

### CODE REVIEW

File: v1/middleware/featureflagstatus.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	MaintenanceStatus = "maintenance"
+	ErrDisabled       = errors.New("Feature Is Disabled")
+	ErrMaintenance    = errors.New("Feature Is Under Maintenance")
+)
+
+type FeatureFlagStatus struct {
+	Status string `json:"status"`
+}
+
+func (mid *Middleware) CheckFeatureFlagStatus(key string) gin.HandlerFunc {
```
Comment (ahtuz) [inline v1/middleware/featureflagstatus.go:25]: titip add penjelasan func ya

File: v1/middleware/featureflagstatus.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	data, err := rest.GetData(body)
+	if err != nil {
+		err = errors.Wrap(err, "get data")
+		return
+	}
+
+	err = json.Unmarshal(data, &status)
+	err = errors.Wrap(err, "unmarshall data")
```
Comment (ahtuz) [inline v1/middleware/featureflagstatus.go:68]:
```suggestion
	err = errors.Wrap(err, "unmarshal data")
```

File: v1/middleware/featureflagstatus.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			rest.ResponseMessage(ctx, http.StatusForbidden, ErrMaintenance.Error())
+			ctx.Abort()
+			return
+		}
+
+		ctx.Next()
+	}
+}
+
+func getFeatureFlagStatus(key string) (status FeatureFlagStatus, err error) {
```
Comment (ahtuz) [inline v1/middleware/featureflagstatus.go:49]: titip add penjelasan func

**Review by ahtuz — CHANGES_REQUESTED** [review]

File: v1/middleware/interface.go [hunk 4]
```diff
@@ -20,5 +20,6 @@ func NewMiddleware(
 type Middlewarer interface {
 	Auth(ctx *gin.Context)
 	CORS(ctx *gin.Context)
+	CheckFeatureFlagStatus(key string)
```
[the remark this answers is not in the document]
Comment (winsonsutanto10) [reply v1/middleware/interface.go:23]: ada disini contextnya
image

**Comment by ahtuz** [comment]
RC pindah ke CU ya

**Review by liemmatthew — APPROVED** [review]

**Review by asianjack19 — APPROVED** [review]

**Review by ahtuz — CHANGES_REQUESTED** [review]
cek CU

**Review by ahtuz — CHANGES_REQUESTED** [review]
rc lagi ngebug gak keluar

**Review by ahtuz — APPROVED** [review]

**Review by ahtuz — CHANGES_REQUESTED** [review]
.

**Review by ahtuz — APPROVED** [review]

**Review by christyyy-a — APPROVED** [review]

**Review by yafetzz — APPROVED** [review]

**Pull request merged**